### PR TITLE
Rust client quickstart updated for 0.12

### PR DESCRIPTION
**Pull request by jdetter**
## Overview

- The purpose of this PR is mainly to get the rust client quickstart up to speed with the new API, we will do another pass on this quickstart for 1.0.

Rendered: https://github.com/clockworklabs/spacetime-docs/blob/200660cbc080e5fba2bb3792e7285f0a0d31f066/docs/sdks/rust/quickstart.md

### CODE REVIEW

File: docs/sdks/rust/quickstart.md [hunk 1]
```diff
@@ -78,132 +80,118 @@ use module_bindings::*;
...
 ```
 
-### Notify about new users
+## Handle errors and disconnections
 
-For each table, we can register on-insert and on-delete callbacks to be run whenever a subscribed row is inserted or deleted. We register these callbacks using the `on_insert` and `on_delete` methods of the trait `TableType`, which is automatically implemented for each table by `spacetime generate`.
+We need to handle connection errors and disconnections by printing appropriate messages and exiting the program.
 
-These callbacks can fire in two contexts:
+To `client/src/main.rs`, add:
```
Comment (jdetter) [inline docs/sdks/rust/quickstart.md:170]:
```suggestion
In `client/src/main.rs`, add:
```

File: docs/sdks/rust/quickstart.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ cargo new client
 Below the `[dependencies]` line in `client/Cargo.toml`, add:
 
 ```toml
-spacetimedb-sdk = "0.7"
+spacetimedb-sdk = "0.12"
```
Comment (gefjon) [inline docs/sdks/rust/quickstart.md:31]: Oh my god, we hadn't updated this in so long...

File: docs/sdks/rust/quickstart.md [hunk 3]
```diff
@@ -78,123 +80,133 @@ use module_bindings::*;
...
-4. When we receive a new message, we'll print it.
-5. When we're informed of the backlog of past messages, we'll sort them and print them in order.
-6. If the server rejects our attempt to set our name, we'll print an error.
-7. If the server rejects a message we send, we'll print an error.
+2. When a new user joins, we'll print a message introducing them.	
+3. When a user is updated, we'll print their new name, or declare their new online status.	
+4. When we receive a new message, we'll print it.	
+5. When we're informed of the backlog of past messages, we'll sort them and print them in order.	
+6. If the server rejects our attempt to set our name, we'll print an error.	
+7. If the server rejects a message we send, we'll print an error.
```
Comment (gefjon) [inline docs/sdks/rust/quickstart.md:120]: Can you do your editor's equivalent of `M-x delete-trailing-whitespace`? There's several instances of trailing whitespace in this file.

File: docs/sdks/rust/quickstart.md [hunk 4]
```diff
@@ -473,15 +469,16 @@ You can find the full code for this client [in the Rust SDK's examples](https://
...
+Our basic terminal interface has some limitations. Incoming messages can appear while the user is typing, which is less than ideal. Additionally, the user's input gets mixed with the program's output, making messages the user sends appear twice. You might want to try improving the interface by using [Rustyline](https://crates.io/crates/rustyline), [Cursive](https://crates.io/crates/cursive), or even creating a full-fledged GUI.
+
+We've tried using Cursive for the interface, and you can check out our implementation in the [Rust SDK's examples](https://github.com/clockworklabs/SpacetimeDB/tree/master/crates/sdk/examples/cursive-chat).
```
Comment (gefjon) [inline docs/sdks/rust/quickstart.md:474]:
```suggestion
```

I didn't bother updating `cursive-chat` for the new SDK, so this is a dead link now.

**Review by gefjon — APPROVED** [review]

**Pull request merged**